mantle/platform/api/openstack/api.go

```diff
@@ -594,7 +595,43 @@ func (a *API) UploadImage(name, path, arch, visibility string) (string, error) {
 	return image.ID, nil
 }
 
-func (a *API) DeleteImage(imageID string) error {
+// TODO: We can remove this block when the follow PR lands:
+//       https://github.com/gophercloud/gophercloud/pull/2221
+type ReplaceImageProtected struct {
+	NewProtected bool
+}
+
+func (r ReplaceImageProtected) ToImagePatchMap() map[string]interface{} {
+	return map[string]interface{}{
+		"op":    "replace",
+		"path":  "/protected",
+		"value": r.NewProtected,
+	}
+}
+
+func (a *API) DeleteImage(imageID string, force bool) error {
+	// Detect if the image is protected from deletion. If protected
+	// and force=true then change protection status and delete it.
+	image, err := images.Get(a.imageClient, imageID).Extract()
+	if err != nil {
+		return err
+	}
```

Comment:
We only actually need this API call if `force` is true, so we could flip things around like this:

```
if force {
  // query protected
  // remove protected
}
// delete
```

And if the API is idempotent, maybe you don't even need to query it first.

Reply:
I guess the open question here is if we care enough to give the user a more specific error message. If we don't query unless forced then the delete will fail (if protected). The user will be able to tell why, but the error message more verbose:

```
Couldn't delete image: Request forbidden: [DELETE https://image.public.mtl1.vexxhost.net/v2/images/b88629d7-660e-42fa-adff-673ec0f6016d], error message: {"message": "Image b88629d7-660e-42fa-adff-673ec0f6016d is protected and cannot be deleted.<br /><br />\n\n\n", "code": "403 Forbidden", "title": "Forbidden"}
```

Reply:
Fair. I guess if we can catch it, then we can mention the force flag which is probably what they'll want to retry with.